### PR TITLE
_filter-based _id retrieval fixes, hapi-fhir-jpaserver-example fix

### PR DESCRIPTION
**Pull request by anthonys123**
(1) Fixes to address _filter-based _id retrievals not being restricted to the specified resource type
(2) Import fix for FhirServerConfig.java in hapi-fhir-jpaserver-example

### CODE REVIEW

**Comment by anthonys123** [comment]
I'll remove the useless null check and push the changes...

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4FilterTest.java [hunk 1]
```diff
@@ -169,6 +169,41 @@ public void testFilterDisabled() {
...
+		String id1 = myPatientDao.create(p).getId().toUnqualifiedVersionless().getValue();
+		String idVal = id1.split("/")[1];
+
+		SearchParameterMap map;
+		List<String> found;
+
+		map = new SearchParameterMap();
+		map.setLoadSynchronous(true);
+		map.add(Constants.PARAM_FILTER, new StringParam(String.format("status eq active or _id eq %s",
+			idVal)));
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4FilterTest.java]: This is a nice touch using the formatter to create the filter string. Makes the test very readable!

**Comment by jamesagnew** [comment]
Looks great, merged.

**Comment by anthonys123** [comment]
Thanks!

**Pull request merged**